### PR TITLE
Add on-start/stop hook methods

### PR DESCRIPTION
**Pull request by pbrisbin**
We're going to be implementing locking around the kafka-offset records. While
we eventually want to move away from this gem entirely, the path of least
resistance at the moment is to add these small hook methods so we can acquire
and release the lock on start and stop respectively.

/cc @codeclimate/review

### CODE REVIEW

File: lib/cc/kafka/consumer.rb [hunk 1]
```diff
@@ -22,15 +22,28 @@ def initialize(client_id, seed_brokers, topic, partition)
...
+
       def on_message(&block)
         @on_message = block
       end
 
       def start
-        trap(:TERM) { stop }
+        trap(:TERM) do
+          @on_stop.call if @on_stop
+          stop
```
Comment (jpignata) [inline lib/cc/kafka/consumer.rb]: Does it make sense to work to ensure all `fetch_message` operations are done before you call the `on_stop` callback? Thinking of a potential race condition depending on what we shove into the callback.

Comment (pbrisbin) [inline lib/cc/kafka/consumer.rb]: Good question. Let me try to re-structure to get that happening.

File: lib/cc/kafka/consumer.rb [hunk 2]
```diff
@@ -22,15 +22,28 @@ def initialize(client_id, seed_brokers, topic, partition)
...
+      end
+
       def on_message(&block)
         @on_message = block
       end
 
       def start
-        trap(:TERM) { stop }
+        trap(:TERM) do
+          @on_stop.call(@offset) if @on_stop
```
Comment (GordonDiggs) [inline lib/cc/kafka/consumer.rb]: Any reason not to put this in `stop` directly?

Comment (pbrisbin) [inline lib/cc/kafka/consumer.rb]: No, but i had to move it anyway; see JP's out of date comment for the reasoning.

**Comment by pbrisbin** [comment]
Ready for re-review.

**Comment by jpignata** [comment]
LGTM

**Comment by GordonDiggs** [comment]
:+1:

**Pull request merged**